### PR TITLE
Fixed mixed domain tests, and use live domain.xml configuration for it.

### PR DESCRIPTION
**Pull request by kabir**
Using the live configuration avoids the eternal problem of these tests
becoming stale. The DomainAdjustors trim the config down into something
more manageable, and make necessary changes to get it to work on the
targetted legacy version. This means that the huge domain.xml file can
now be removed, and only one host-slave.xml file is needed.

While here, I have changed the structure since I was never happy with
it, it made it hard to add tests for new versions. This new structure is
clearer to me anyway.

It also removes testing for legacy version we don't care about, and adds it for legacy versions we care about.

The infinispan and jgroups transformers are not working for this scenario, so I have created https://issues.jboss.org/browse/WFLY-4515 to make sure this gets fixed for 9.0.0.GA.
https://issues.jboss.org/browse/WFLY-4435 is also vaguely related.

### CODE REVIEW

**Comment by kabir** [comment]
Mixed domain test run: http://brontes.lab.eng.brq.redhat.com/viewLog.html?buildId=50037&tab=buildResultsDiv&buildTypeId=WF_MasterLinuxMixedDomain (I turned off the -Djboss.test.transformers.eap option for this run, since we need https://issues.jboss.org/browse/WFLY-4515)

File: testsuite/mixed-domain/src/test/java/org/jboss/as/test/integration/domain/mixed/MixedDomainTestSuite.java [hunk 1]
```diff
@@ -51,18 +50,24 @@
...
+        final Version.AsVersion version = getVersion(testClass);
         if (support == null) {
+            final MixedDomainTestSupport testSupport;
             try {
-                final MixedDomainTestSupport testSupport = MixedDomainTestSupport.create(testClass.getSimpleName(), version);
-                // Start!
+                testSupport = MixedDomainTestSupport.create(testClass.getSimpleName(), version);
+            } catch (Exception e) {
+                e.printStackTrace();
+                throw new RuntimeException(e);
```
Comment (ctomc) [inline testsuite/mixed-domain/src/test/java/org/jboss/as/test/integration/domain/mixed/MixedDomainTestSuite.java]: printstacktrace and exception throwing

File: testsuite/mixed-domain/src/test/java/org/jboss/as/test/integration/domain/mixed/MixedDomainTestSupport.java [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+        File file = new File(".");
+        for (String part : parts) {
+            file = new File(file, part);
+            Assert.assertTrue(file.getAbsolutePath() + " does not exist", file.exists());
+        }
+        return file;
+    }
+
+    private static File createDirectory(String...parts) {
+        File file = new File(".");
```
Comment (ctomc) [inline testsuite/mixed-domain/src/test/java/org/jboss/as/test/integration/domain/mixed/MixedDomainTestSupport.java]: pretty much whole method can be replaced by 
Files.createDirectories(Paths.get(parts))

File: testsuite/mixed-domain/src/test/java/org/jboss/as/test/integration/domain/mixed/MixedDomainTestSupport.java [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+        return "copied-master-config/domain.xml";
+    }
+
+    private static File loadFile(String... parts) {
+        File file = new File(".");
+        for (String part : parts) {
+            file = new File(file, part);
+            Assert.assertTrue(file.getAbsolutePath() + " does not exist", file.exists());
+        }
+        return file;
```
Comment (ctomc) [inline testsuite/mixed-domain/src/test/java/org/jboss/as/test/integration/domain/mixed/MixedDomainTestSupport.java]: could be replaced with:
Path p =  Paths.get(parts);
Files.exists(p)
return p.toFile(); //if you insist on using java.io.File

File: testsuite/mixed-domain/src/test/java/org/jboss/as/test/integration/domain/mixed/Version.java [hunk 4]
```diff
@@ -41,9 +41,10 @@
     final String EAP = "jboss-eap-";
 
     enum AsVersion {
-        AS_7_1_2_FINAL(AS, "7.1.2.Final"),
-        AS_7_1_3_FINAL(AS, "7.1.3.Final"),
-        AS_7_2_0_FINAL(AS, "7.2.0.Final");
+        AS_7_2_0_FINAL(AS, "7.2.0.Final"),
```
Comment (ctomc) [inline testsuite/mixed-domain/src/test/java/org/jboss/as/test/integration/domain/mixed/Version.java]: we can drop 7.2.0. as well, first we support is EAP 6.2 aka 7.3

File: testsuite/mixed-domain/src/test/java/org/jboss/as/test/integration/domain/mixed/eap630/DomainAdjuster630.java [hunk 5]
```diff
@@ -0,0 +1,345 @@
...
+import org.wildfly.extension.batch.BatchSubsystemExtension;
+import org.wildfly.extension.beanvalidation.BeanValidationExtension;
+import org.wildfly.extension.io.IOExtension;
+import org.wildfly.extension.requestcontroller.RequestControllerExtension;
+import org.wildfly.extension.security.manager.SecurityManagerExtension;
+import org.wildfly.extension.undertow.UndertowExtension;
+import org.wildfly.iiop.openjdk.IIOPExtension;
+
+/**
+ * Does adjustments to the domain model for 7.2.0 legacy slaves
```
Comment (ctomc) [inline testsuite/mixed-domain/src/test/java/org/jboss/as/test/integration/domain/mixed/eap630/DomainAdjuster630.java]: for EAP 6.3.0 legacy slaves

**Comment by bstansberry** [comment]
Looks good.

**Comment by kabir** [comment]
Another mixed domain run with the latest fixes http://brontes.lab.eng.brq.redhat.com/viewLog.html?buildId=50059&tab=buildResultsDiv&buildTypeId=WF_MasterLinuxMixedDomain

**Comment by bstansberry** [comment]
Kabir, please go ahead and merge this when you are ready.

**Pull request merged**